src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -1689,6 +1689,17 @@ private static bool GetIsWindows7Or2008R2()
             return false;
         }
 
+        internal void HeartBeat()
+        {
+            Http2Connection[]? localHttp2Connections = _http2Connections;
```

Comment:
This variable is modified concurrently.

```suggestion
            Http2Connection[]? localHttp2Connections = Volatile.Read(ref _http2Connections);
```

Reply:
Pointer read is atomic, isn't it? _http2Connections is already volatile, so memory barrier should be in place.